apps/fares/lib/transfer.ex

```diff
@@ -0,0 +1,120 @@
+defmodule Fares.Transfer do
+  @moduledoc """
+    Tools for handling logic around transfers between transit legs and modes.
+    The MBTA allows transfers between services depending on the fare media used
+    and the amount paid.
+
+    This logic may be superseded by the upcoming fares work.
+  """
+  require Fares
+  alias Routes.{Repo, Route}
+  alias TripPlan.Leg
+
+  defguard has_nil(pair)
+           when is_nil(elem(pair, 0)) or
+                  is_nil(elem(pair, 1))
+
+  @type fare_atom :: Route.gtfs_route_type() | :inner_express_bus | :outer_express_bus
+
+  # Paying a single-ride fare for the first may get you a transfer to the second
+  # (can't be certain, as it depends on media used)!
+  @single_ride_valid_transfers [
+    [:subway, :subway],
+    [:subway, :bus],
+    [:bus, :subway],
+    [:bus, :bus],
+    [:inner_express_bus, :subway],
+    [:outer_express_bus, :subway],
+    [:inner_express_bus, :bus],
+    [:outer_express_bus, :bus]
+  ]
+
+  # Our stops don't model underground transfer points, but the system is small
+  # enough that we can just hardcode these for now. Note: unknown directionality
+  @underground_xfers [
+    %{lines: ["Red", "Green-B", "Green-C", "Green-D", "Green-E"], stop: "place-pktrm"},
+    %{lines: ["Red", "Orange"], stop: "place-dwnxg"},
+    %{lines: ["Orange", "Blue"], stop: "place-state"},
+    %{lines: ["Blue", "Green-C", "Green-D", "Green-E"], stop: "place-gover"},
+    %{lines: ["Red", "741", "742", "743"], stop: "place-sstat"},
+    %{lines: ["Orange", "Green-C", "Green-E"], stop: "place-north"},
+    %{lines: ["Orange", "Green-C", "Green-E"], stop: "place-haecl"},
+    %{lines: ["Green-B", "Green-C", "Green-D"], stop: "place-kencl"},
+    %{lines: ["Green-B", "Green-C", "Green-D", "Green-E"], stop: "place-armnl"}
+  ]
+
+  @doc """
+  Takes a pair of routes and returns true if there might be a transfer between
+  the two, based on the list in @single_ride_valid_transfers
+  """
+  @spec is_maybe_transfer([Route.id_t()]) :: boolean | nil
```

Comment:
Nit: the [convention](https://hexdocs.pm/elixir/master/naming-conventions.html#trailing-question-mark-foo) in Elixir is to name functions that return booleans with a trailing question mark. (Same thing for `is_free_transfer`.)